lisa/tests/staging/sched_android.py

```diff
@@ -0,0 +1,239 @@
+        """
+        return super().from_target(target, res_dir, ftrace_coll=ftrace_coll)
+
+    @classmethod
+    def _from_target(cls, target, res_dir, ftrace_coll):
+        return cls(res_dir, target.plat_info,
+            list(cls._create_test_bundles(target, res_dir, ftrace_coll))
+        )
+
+    @classmethod
+    @abc.abstractmethod
+    def _create_test_bundles(cls, target, res_dir, ftrace_coll):
+        """
+        Collects and returns a ResultBundle per test item.
+        """
+        pass
+
+    @classmethod
+    def _create_test_bundle(cls, target, res_dir, ftrace_coll, item_cls, boost, pi):
+        """
+        Creates and returns a TestBundle for a given item class, and a given
+        schedtune configuration
+        """
+        item_dir = os.path.join(res_dir, 'boost_{}'.format(boost))
+        os.makedirs(item_dir)
+
+        logger = cls.get_logger()
+        logger.info('Running {} with boost={}, prefer_idle={}'.format(
+                    item_cls.__name__, boost, pi))
+        return item_cls.from_target(target, boost, pi, res_dir=item_dir, ftrace_coll=ftrace_coll)
+
+    def _merge_res_bundles(self, res_bundles):
+        """
+        Merge a set of result bundles
+        """
+        overall_bundle = ResultBundle.from_bool(all(res_bundles.values()))
+        for name, bundle in res_bundles.items():
+            overall_bundle.add_metric(name, bundle.metrics)
+
+        overall_bundle.add_metric('failed', [
+            name for name, bundle in res_bundles.items()
+            if bundle.result is Result.FAILED
+        ])
+        return overall_bundle
+
+class SchedTuneFreqItem(SchedTuneItemBase):
+    """
+    Runs a tiny RT rtapp task pinned to a big CPU at a given boost level and
+    checks the frequency selection was performed accordingly.
+    """
+
+    ftrace_conf = FtraceConf({
+        "events" : ["sched_switch", "cpu_frequency"]
+    }, __qualname__)
+
+    @classmethod
+    def get_rtapp_profile(cls, plat_info):
+        cpu = plat_info['capacity-classes'][-1][0]
+        rtapp_profile = {}
+        rtapp_profile['rta_stune'] = Periodic(
+            duty_cycle_pct = 1, # very small task, no impact on freq w/o boost
+            duration_s = 10,
+            period_ms = 16,
+            cpus = [cpu], # pin to big CPU, to focus on frequency selection
+            sched_policy = 'FIFO' # RT tasks have the boost holding feature so
+                                  # the frequency should be more stable, and we
+                                  # shouldn't go to max freq in Android
+        )
+        return rtapp_profile
+
+    def trace_window(self, trace):
+        """
+        Set the boundaries of the trace window to ``cpu_frequency`` events
+        before/after the task's start/end time
+        """
+        rta_start, rta_stop = super().trace_window(trace)
+
+        cpu = self.plat_info['capacity-classes'][-1][0]
+        fdf = trace.df_events('cpu_frequency')
```

Comment:
Don't we have an informal code convention to call DF related symbols with a `_df` suffix?